### ISSUE TITLE
New linter: detect throw in finally

### ISSUE THREAD

**Issue by marco-carvalho** [clj-kondo/clj-kondo#2970]
- [ ] I have read the [Clojure etiquette](https://clojure.org/community/etiquette) and will respect it when communicating on this platform.

**Is your feature request related to a problem? Please describe.**

Throwing directly from a `finally` body replaces a pending return value or exception. This can silently hide the original failure and is easy to overlook in nested forms.

```clojure
(try
  (work)
  (finally
    (throw (Exception. "cleanup failed"))))
```

**Describe the solution you'd like**

Add a `:throw-in-finally` linter, enabled at warning level by default, for direct throws within the current `finally` execution scope. A throw inside a nested function definition should not warn because it runs in a different scope. Generated forms should not warn.

**Describe alternatives you've considered**

A broad warning for all potentially throwing calls in `finally` would produce many false positives. Restricting the check to explicit `throw` forms keeps it predictable.

**Additional context**

I am interested in submitting a focused PR with nested forms and function-boundary tests if this approach is accepted.
